lib/manageiq/api/client/error.rb

```diff
@@ -0,0 +1,31 @@
+module ManageIQ
+  module API
+    module Client
+      class Error
+        attr_accessor :status
+        attr_accessor :kind
+        attr_accessor :message
+        attr_accessor :klass
+
+        def initialize(status = 0, json_response = {})
+          update(status, json_response)
+        end
+
+        def clear
+          update(0)
+        end
+
+        def update(status, json_response = {})
+          @status = status
+          @kind, @message, @klass = nil
```

Comment:
Interesting!  I've never seen it coded this way...I've always seen `@kind = @message = @klass = nil`  I wonder if there is a performance difference between the two.